client/src/http_default.rs

```diff
@@ -0,0 +1,199 @@
+use std::{borrow::Borrow, net::TcpStream};
+
+use attohttpc::{
+    body as atto_body, RequestBuilder as AttoHttpRequestBuilder, Response as AttoHttpResponse,
+};
+use eyre::{eyre, Error, Result, WrapErr};
+use http::header::HeaderName;
+use tungstenite::{stream::MaybeTlsStream, WebSocket};
+pub use tungstenite::{Error as WebSocketError, Message as WebSocketMessage};
+
+use crate::http::{Headers, Method, RequestBuilder, Response};
+
+type Bytes = Vec<u8>;
+type AttoHttpRequestBuilderWithBytes = AttoHttpRequestBuilder<atto_body::Bytes<Bytes>>;
+
+trait SetSingleHeader {
+    fn header(self, key: HeaderName, value: String) -> Self;
+}
+
+impl SetSingleHeader for AttoHttpRequestBuilderWithBytes {
+    fn header(self, key: HeaderName, value: String) -> Self {
+        self.header(key, value)
+    }
+}
+
+impl SetSingleHeader for http::request::Builder {
+    fn header(self, key: HeaderName, value: String) -> Self {
+        self.header(key, value)
+    }
+}
+
+trait SetHeadersExt: Sized + SetSingleHeader {
+    fn set_headers(mut self, headers: Headers) -> Result<Self> {
```

Comment:
Do not `#[inline]` this function is not trivial at the assembly level because of the `wrap_err_with`.